### PR TITLE
8295944: Move the Http2TestServer and related classes into a package of its own

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review of this change which addresses https://bugs.openjdk.org/browse/JDK-8295944?

As noted in the issue, the HttpClient testing helper infrastructure code resides in a default (unnamed) packages within the test hierarchy. The commit in this PR moves those classes into a specific package. The tests have been updated to refer to them correctly with proper package name references.
Additionally as noted in the issue, the `@modules` declaration used in the httpclient tests, were all the same and were repeated in all these tests. The commit in this  PR moves those module references into the `TEST.properties` to centralize such reference. Any test which needs additional modules can then use specific `@modules` declaration in the test definition.

No functional changes have been done in this PR.

tier1, tier2 and tier3 testing passed after these changes.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295944](https://bugs.openjdk.org/browse/JDK-8295944): Move the Http2TestServer and related classes into a package of its own

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12160/head:pull/12160` \
`$ git checkout pull/12160`

Update a local copy of the PR: \
`$ git checkout pull/12160` \
`$ git pull https://git.openjdk.org/jdk pull/12160/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12160`

View PR using the GUI difftool: \
`$ git pr show -t 12160`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12160.diff">https://git.openjdk.org/jdk/pull/12160.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/net/httpclient/FlowAdapterSubscriberTest.java [hunk 1]
```diff
@@ -62,15 +66,10 @@
 /*
  * @test
  * @summary Basic tests for Flow adapter Subscribers
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/FlowAdapterSubscriberTest.java:70]:
```suggestion
```

File: test/jdk/java/net/httpclient/ExecutorShutdown.java [hunk 2]
```diff
@@ -26,15 +26,11 @@
  * @bug 8277969
  * @summary Test for edge case where the executor is not accepting
  *          new tasks while the client is still running
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/ExecutorShutdown.java:30]:
```suggestion
```

File: test/jdk/java/net/httpclient/CookieHeaderTest.java [hunk 3]
```diff
@@ -25,15 +25,10 @@
  * @test
  * @bug 8199851
  * @summary Test for multiple vs single cookie header for HTTP/2 vs HTTP/1.1
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/CookieHeaderTest.java:29]:
```suggestion
```

File: test/jdk/java/net/httpclient/HeadTest.java [hunk 4]
```diff
@@ -25,15 +25,10 @@
  * @test
  * @bug 8203433 8276559
  * @summary (httpclient) Add tests for HEAD and 304 responses.
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/HeadTest.java:29]:
```suggestion
```

File: test/jdk/java/net/httpclient/PathSubscriber/BodySubscriberOfFileTest.java [hunk 5]
```diff
@@ -27,18 +27,16 @@
  * @summary Confirm HttpResponse.BodySubscribers#ofFile(Path)
  *          works with default and non-default file systems
  *          when SecurityManager is enabled
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/PathSubscriber/BodySubscriberOfFileTest.java:30]:
```suggestion
```

File: test/jdk/java/net/httpclient/RedirectWithCookie.java [hunk 6]
```diff
@@ -24,15 +24,10 @@
 /*
  * @test
  * @summary Test for cookie handling when redirecting
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/RedirectWithCookie.java:28]:
```suggestion
```

File: test/jdk/java/net/httpclient/Response1xxTest.java [hunk 7]
```diff
@@ -49,14 +50,11 @@
  * @test
  * @bug 8292044
  * @summary Tests behaviour of HttpClient when server responds with 102 or 103 status codes
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/Response1xxTest.java:54]:
```suggestion
```

File: test/jdk/java/net/httpclient/RetryWithCookie.java [hunk 8]
```diff
@@ -25,15 +25,12 @@
  * @test
  * @bug 8199943
  * @summary Test for cookie handling when retrying after close
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/RetryWithCookie.java:29]:
```suggestion
```

File: test/jdk/java/net/httpclient/StreamingBody.java [hunk 9]
```diff
@@ -25,15 +25,10 @@
  * @test
  * @summary Exercise a streaming subscriber ( InputStream ) without holding a
  *          strong (or any ) reference to the client.
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/StreamingBody.java:29]:
```suggestion
```

File: test/jdk/java/net/httpclient/UnauthorizedTest.java [hunk 10]
```diff
@@ -29,15 +29,10 @@
  *      header only in the case where an authenticator is configured
  *      for the client. If no authenticator is configured the client
  *      should simply let the caller deal with the unauthorized response.
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/UnauthorizedTest.java:33]:
```suggestion
```

File: test/jdk/java/net/httpclient/UserCookieTest.java [hunk 11]
```diff
@@ -26,15 +26,10 @@
  * @bug 8276774
  * @summary Test that user-supplied cookies are appended to
  *          server-cookies for HTTP/2 vs HTTP/1.1
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/UserCookieTest.java:30]:
```suggestion
```

File: test/jdk/java/net/httpclient/BasicRedirectTest.java [hunk 12]
```diff
@@ -24,15 +24,10 @@
 /*
  * @test
  * @summary Basic test for redirect and redirect policies
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/BasicRedirectTest.java:28]:
```suggestion
```

File: test/jdk/java/net/httpclient/FilePublisher/FilePublisherPermsTest.java [hunk 13]
```diff
@@ -29,14 +29,10 @@
  *          policy 1: no custom permission
  *          policy 2: custom permission for test classes
  *          policy 3: custom permission for test classes and httpclient
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/FilePublisher/FilePublisherPermsTest.java:32]:
```suggestion
```

File: test/jdk/java/net/httpclient/FilePublisher/FilePublisherTest.java [hunk 14]
```diff
@@ -26,14 +26,10 @@
  * @bug 8235459
  * @summary Confirm that HttpRequest.BodyPublishers#ofFile(Path)
  *          assumes the default file system
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/FilePublisher/FilePublisherTest.java:29]:
```suggestion
```

File: test/jdk/java/net/httpclient/MaxStreams.java [hunk 15]
```diff
@@ -26,15 +26,10 @@
  * @bug 8196389
  * @summary Should HttpClient support SETTINGS_MAX_CONCURRENT_STREAMS from the server
  *
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          java.logging
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/MaxStreams.java:30]:
```suggestion
```

File: test/jdk/java/net/httpclient/PathSubscriber/BodyHandlerOfFileDownloadTest.java [hunk 16]
```diff
@@ -26,18 +26,18 @@
  * @bug 8237470
  * @summary Confirm HttpResponse.BodySubscribers#ofFileDownload(Path)
  *          works only with the default file system
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/PathSubscriber/BodyHandlerOfFileDownloadTest.java:29]:
```suggestion
```

File: test/jdk/java/net/httpclient/PathSubscriber/BodyHandlerOfFileTest.java [hunk 17]
```diff
@@ -27,18 +27,17 @@
  * @summary Confirm HttpResponse.BodyHandlers#ofFile(Path)
  *          works with default and non-default file systems
  *          when SecurityManager is enabled
- * @modules java.base/sun.net.www.http
- *          java.net.http/jdk.internal.net.http.common
- *          java.net.http/jdk.internal.net.http.frame
- *          java.net.http/jdk.internal.net.http.hpack
  *          jdk.httpserver
```
Comment (dfuch) [inline test/jdk/java/net/httpclient/PathSubscriber/BodyHandlerOfFileTest.java:30]:
```suggestion
```

**Review by dfuch — CHANGES_REQUESTED** [review]
Very nice refactoring overall. Thanks for taking that on! There are a few places where some lines should have been removed but were not. I made some suggestions. Also there are 9 classes that have kept their custom `@modules` clause because they want to access 
```
 *          java.base/sun.net.www
 *          java.base/sun.net 
 ```
 I wonder if we should just add these two in TEST.properties - and then remove the `@modules` from these nine classes as well?

**Comment by jaikiran** [comment]
Hello Daniel,

> There are a few places where some lines should have been removed but were not. I made some suggestions.

Thank you for catching these. These were genuine issues and were a leftover from my `@modules` removal. I've fixed these now in the updated PR. I went through all the files again and did a search (with a different search term this time) and I think have now covered all these leftovers.

**Comment by jaikiran** [comment]
> Also there are 9 classes that have kept their custom @modules clause because they want to access
>
>           java.base/sun.net.www
>           java.base/sun.net 
>
> I wonder if we should just add these two in TEST.properties - and then remove the @modules from these nine classes as well?

I think that's reasonable. When I changed most of these tests to remove `@modules` from their test definition, my selection of "common" declarations missed out these two declarations. I've now updated the `TEST.properties` to include these and also updated the relevant tests to remove the `@modules` declaration.

After this change, there remains few tests which use only a limited subset of the modules delcared in `TEST.properties`. I have left those tests as-is (for example `CancelledResponse.java`), since it didn't look necessary to remove those `@modules`. Let me know if you prefer otherwise and I'll update them as well.

Local testing with the latest state of the PR passed. I've now triggered a broader testing with these changes.

**Review by dfuch — APPROVED** [review]

**Comment by jaikiran** [comment]
Thank you Daniel for the review. tier1, tier2, tier3 testing came back successful.

**Pull request closed without merging**